cleanlab/datalab/issue_manager/imbalance.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (C) 2017-2023  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, ClassVar, Dict, List, Optional, Union
+import warnings
+
+import numpy as np
+import pandas as pd
+from cleanlab.datalab.issue_manager import IssueManager
+
+if TYPE_CHECKING:  # pragma: no cover
+    import numpy.typing as npt
+    from cleanlab.datalab.datalab import Datalab
+
+
+class ClassImbalanceIssueManager(IssueManager):
+    """Manages issues related to imbalance class examples."""
+
+    description: ClassVar[
+        str
+    ] = """Examples belonging to the most under-represented class in the dataset.
+
+    Parameters
+    ----------
+    datalab :
+        The Datalab instance that this issue manager searches for issues in.
+
+    fraction:
+        Minimum fraction of samples of each class that are present in a dataset without class imbalance.
+    """
+    issue_name: ClassVar[str] = "class_imbalance"
+    verbosity_levels = {
+        0: [],
+        1: [],
+        2: ["threshold"],
```

Comment:
So I understand correctly, the threshold is related to the `fraction: float = 0.1` parameter?

Do you intend to include it in the report?

Reply:
Good point, I am actually not sure. Would you have any suggestions for this? (When I pushed the code I didn't completely understand the function of verbosity levels).

Reply:
For now, you can just 


1. Rename `fraction` -> `threshold`
2. Update the class docstring accordingly.
3. Remove `threshold` from the verbosity_levels for now. This only matters when running `Datalab.report()`.
4. `find_issues` should set the `self.info` dictionary. You could add a `{"threshold": threshold}` to `self.info`. We usually extract the code to the `collect_info` method ([see docs](https://docs.cleanlab.ai/stable/cleanlab/datalab/issue_manager/issue_manager.html#cleanlab.datalab.issue_manager.issue_manager.IssueManager.collect_info)). Below is an example of how this is done in the `NearDuplicatesIssueManager`.

https://github.com/cleanlab/cleanlab/blob/1575ba743231dc741076cfc6c37f195f27e80442/cleanlab/datalab/issue_manager/duplicate.py#L113
https://github.com/cleanlab/cleanlab/blob/1575ba743231dc741076cfc6c37f195f27e80442/cleanlab/datalab/issue_manager/duplicate.py#L159-L188